src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,200 @@
+
+  public IntakeIOTalonFX() {
+    rightRollerIRSensor = new DigitalInput(IntakeConstants.INTAKE_RIGHT_ROLLER_IR_SENSOR_ID);
+    leftRollerIRSensor = new DigitalInput(IntakeConstants.INTAKE_LEFT_ROLLER_IR_SENSOR_ID);
+    drumIRSensor = new DigitalInput(IntakeConstants.INTAKE_DRUM_IR_SENSOR_ID);
+
+    rightRollerVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    leftRollerVelocityRequest = new VelocityTorqueCurrentFOC(0);
+    drumVelocityRequest = new VelocityTorqueCurrentFOC(0);
+
+    rightRollerVelocityStatusSignal = rightRollerMotor.getRotorVelocity();
+    leftRollerVelocityStatusSignal = leftRollerMotor.getRotorVelocity();
+    drumVelocityStatusSignal = drumMotor.getRotorVelocity();
+
+    rightRollerStatorCurrentStatusSignal = rightRollerMotor.getStatorCurrent();
+    leftRollerStatorCurrentStatusSignal = leftRollerMotor.getStatorCurrent();
+    drumStatorCurrentStatusSignal = drumMotor.getStatorCurrent();
+
+    rightRollerSupplyCurrentStatusSignal = rightRollerMotor.getSupplyCurrent();
+    leftRollerSupplyCurrentStatusSignal = leftRollerMotor.getSupplyCurrent();
+    drumSupplyCurrentStatusSignal = drumMotor.getSupplyCurrent();
+
+    configureIntakeRollerMotors(
+        IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_ID, IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_ID);
+    configureIntakeDrumMotor(IntakeConstants.INTAKE_DRUM_MOTOR_ID);
+  }
+
+  @Override
+  public void updateInputs(IntakeIOInputs inputs) {
+    BaseStatusSignal.refreshAll(
+        rightRollerVelocityStatusSignal,
+        leftRollerVelocityStatusSignal,
+        drumVelocityStatusSignal,
+        rightRollerStatorCurrentStatusSignal,
+        leftRollerStatorCurrentStatusSignal,
+        drumStatorCurrentStatusSignal,
+        rightRollerSupplyCurrentStatusSignal,
+        leftRollerSupplyCurrentStatusSignal,
+        drumSupplyCurrentStatusSignal);
+
+    inputs.isRightRollerIRBlocked = rightRollerIRSensor.get();
+    inputs.isLeftRollerIRBlocked = leftRollerIRSensor.get();
+    inputs.isDrumIRBlocked = drumIRSensor.get();
+
+    inputs.rightRollerStatorCurrentAmps = rightRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerStatorCurrentAmps = leftRollerStatorCurrentStatusSignal.getValueAsDouble();
+    inputs.drumStatorCurrentAmps = drumStatorCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerSupplyCurrentAmps = rightRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.leftRollerSupplyCurrentAmps = leftRollerSupplyCurrentStatusSignal.getValueAsDouble();
+    inputs.drumSupplyCurrentAmps = drumSupplyCurrentStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerVelocityRotationsPerSecond =
+        rightRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.leftRollerVelocityRotationsPerSecond = leftRollerVelocityStatusSignal.getValueAsDouble();
+    inputs.drumVelocityRotationsPerSecond = drumVelocityStatusSignal.getValueAsDouble();
+
+    inputs.rightRollerReferenceVelocityRPS = rightRollerRequestedVelocity;
```

Comment:
Get the reference value from the device instead of what we specified to this object. For example:
```
inputs.rightRollerReferenceVelocityRPS = rightRollerMotor.getClosedLoopReference().getValue();
```
(You could make status signal objects for this. Honestly, I'm not sure when it's worthwhile to make status signal objects. While it does allow you to do a single refresh, I'm not sure how much time that saves. Probably best to make status signal objects for everything?)